src/main/java/io/kestra/plugin/aws/runner/AwsBatchScriptRunner.java

```diff
@@ -0,0 +1,546 @@
+import io.kestra.core.models.annotations.PluginProperty;
+import io.kestra.core.models.script.*;
+import io.kestra.core.runners.RunContext;
+import io.kestra.core.utils.Await;
+import io.kestra.core.utils.IdUtils;
+import io.kestra.plugin.aws.AbstractConnectionInterface;
+import io.kestra.plugin.aws.s3.AbstractS3;
+import io.micronaut.core.annotation.Introspected;
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.validation.constraints.NotNull;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+import org.reactivestreams.Subscription;
+import org.slf4j.Logger;
+import reactor.core.CoreSubscriber;
+import software.amazon.awssdk.http.apache.ApacheHttpClient;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.batch.BatchClient;
+import software.amazon.awssdk.services.batch.BatchClientBuilder;
+import software.amazon.awssdk.services.batch.model.*;
+import software.amazon.awssdk.services.cloudwatchlogs.CloudWatchLogsAsyncClient;
+import software.amazon.awssdk.services.cloudwatchlogs.model.*;
+import software.amazon.awssdk.services.s3.S3AsyncClient;
+import software.amazon.awssdk.services.s3.model.*;
+import software.amazon.awssdk.transfer.s3.S3TransferManager;
+import software.amazon.awssdk.transfer.s3.model.DownloadFileRequest;
+import software.amazon.awssdk.transfer.s3.model.FileDownload;
+import software.amazon.awssdk.transfer.s3.model.FileUpload;
+import software.amazon.awssdk.transfer.s3.model.UploadFileRequest;
+
+import java.net.URI;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static io.kestra.core.utils.Rethrow.throwConsumer;
+
+@Introspected
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Beta
+@Schema(title = "AWS Batch script runner", description = """
+    Run a script in a container on an AWS Batch Compute Environment.
+    To use `inputFiles`, `outputFiles` and `namespaceFiles` properties, you must provide a `s3Bucket`.
+    This runner will wait for the task to succeed or fail up to a max `waitUntilCompletion` duration.
+    It will return with an exit code according to the following mapping:
+    - SUCCEEDED: 0
+    - FAILED: 1
+    - RUNNING: 2
+    - RUNNABLE: 3
+    - PENDING: 4
+    - STARTING: 5
+    - SUBMITTED: 6
+    - OTHER: -1""")
+public class AwsBatchScriptRunner extends ScriptRunner implements AbstractS3, AbstractConnectionInterface {
+    private static final Map<JobStatus, Integer> exitCodeByStatus = Map.of(
+        JobStatus.FAILED, 1,
+        JobStatus.RUNNING, 2,
+        JobStatus.RUNNABLE, 3,
+        JobStatus.PENDING, 4,
+        JobStatus.STARTING, 5,
+        JobStatus.SUBMITTED, 6,
+        JobStatus.UNKNOWN_TO_SDK_VERSION, -1
+    );
+
+    @NotNull
+    @PluginProperty(dynamic = true)
```

Comment:
All those comes from the interface so no need to add them here (the same for all properties that comes from the AbstractConnectionInterface)